packages/patternfly-4/react-core/src/components/TextInput/examples/FocusTextInput.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { TextInput } from '@patternfly/react-core';
+
+class FocusTextInput extends React.Component {
+  static title = 'TextInput Focus';
+
+  state = {
+    value: ''
+  };
+
+  constructor(props) {
+    super(props);
+    this.inputRef = null;
+  }
+
+  handleTextInputChange = value => {
+    this.setState({ value });
+  };
+
+  componentDidUpdate() {
+    this.inputRef && this.inputRef.focus();
+  }
+
+  render() {
+    const { value } = this.state;
+    const FocusTextInput = React.forwardRef(({ value, type, onChange }, ref) => (
```

Comment:
This is actually not how this is meant to be used. Forward ref should wrap the actual TextInput component that we export. That would allow for ref passed instead of innerRef

Reply:
I am not sure I understand.

If I do 
```js
React.forwardRef((props, ref) => <TextInput ref={ref} {...props} />)
```
the `TextInput` will be referenced instead of the `input` node.

At any rate, I couldn't come up with any way to pass `ref` without using `innerRef`.